### DISCUSSION TITLE
Maximum size of image exceeded when trying to redact

### DISCUSSION THREAD

**Discussion by buzzman2** [Stirling-Tools/Stirling-PDF#1605] [Q&A] [answered]
Trying to redact stuff from a pdf that is 4.07 mb and I get:
15:10:42.989 [qtp1915898213-38] WARN  o.e.j.ee10.servlet.ServletChannel - handleException /api/v1/security/auto-redact java.io.IOException: Maximum size of image exceeded (w * h * scale ^ 2) = 9062.0 * 17500.0 * 4.1666665 ^ 2 > 2147483647

I am guessing it does not like the size of the pdf? Is there anyway to increase this?

**Comment by Frooodle** [comment]
Looks like the resolution of the image is what's to large, either a image within the pdf or a single pdf page is to big and hence the pdf image (assuming you have to convert to image part enabled) is to large 

Any chance you can send the PDF or send the JSON file from get pdf info function in Stirling pdf

**Reply by Frooodle** [reply]
Also full stack trace of log can help too

**Reply by buzzman2** [reply]
Thanks. Unfortunately, it is a credit card statement. I have done a few and they seemed to work but not this one. Am I just copying the stack trace form the dos window? How can I send the two files? Sorry, first time using github

**Comment by buzzman2** [comment]
____ _____ ___ ____  _     ___ _   _  ____       ____  ____  _____
/ ___|_   _|_ _|  _ \| |   |_ _| \ | |/ ___|     |  _ \|  _ \|  ___|
\___ \ | |  | || |_) | |    | ||  \| | |  _ _____| |_) | | | | |_
 ___) || |  | ||  _ <| |___ | || |\  | |_| |_____|  __/| |_| |  _|
|____/ |_| |___|_| \_\_____|___|_| \_|\____|     |_|   |____/|_|
Powered by Spring Boot 3.2.4
15:52:55.819 [main] INFO  s.software.SPDF.SPdfApplication - Starting SPdfApplication v0.26.1 using Java 21.0.4 with PID 11232 (C:\Users\user\Downloads\Stirling-PDF-with-login.exe started by user in C:\Users\user\Downloads)
15:52:55.823 [main] INFO  s.software.SPDF.SPdfApplication - No active profile set, falling back to 1 default profile: "default"
15:52:57.754 [main] INFO  o.s.d.r.c.RepositoryConfigurationDelegate - Bootstrapping Spring Data JPA repositories in DEFAULT mode.
15:52:57.864 [main] INFO  o.s.d.r.c.RepositoryConfigurationDelegate - Finished Spring Data repository scanning in 95 ms. Found 3 JPA repository interfaces.
15:52:59.475 [main] INFO  o.s.b.w.e.j.JettyServletWebServerFactory - Server initialized with port: 8080
15:52:59.489 [main] INFO  org.eclipse.jetty.server.Server - jetty-12.0.7; built: 2024-02-29T21:19:41.771Z; git: c89aca8fd34083befd79f328a3b8b6ffff04347e; jvm 21.0.4+8-LTS-274
15:52:59.545 [main] INFO  o.e.j.s.h.ContextHandler.application - Initializing Spring embedded WebApplicationContext
15:52:59.547 [main] INFO  o.s.b.w.s.c.ServletWebServerApplicationContext - Root WebApplicationContext: initialization completed in 3637 ms
15:53:00.176 [main] INFO  o.h.jpa.internal.util.LogHelper - HHH000204: Processing PersistenceUnitInfo [name: default]
15:53:00.314 [main] INFO  org.hibernate.Version - HHH000412: Hibernate ORM core version 6.4.4.Final
15:53:00.393 [main] INFO  o.h.c.i.RegionFactoryInitiator - HHH000026: Second-level cache disabled
15:53:00.923 [main] INFO  o.s.o.j.p.SpringPersistenceUnitInfo - No LoadTimeWeaver setup: ignoring JPA class transformer
15:53:00.984 [main] INFO  com.zaxxer.hikari.HikariDataSource - HikariPool-1 - Starting...
15:53:01.250 [main] INFO  com.zaxxer.hikari.pool.HikariPool - HikariPool-1 - Added connection conn0: url=jdbc:h2:file:./configs/stirling-pdf-DB user=SA
15:53:01.256 [main] INFO  com.zaxxer.hikari.HikariDataSource - HikariPool-1 - Start completed.
15:53:02.751 [main] INFO  o.h.e.t.j.p.i.JtaPlatformInitiator - HHH000489: No JTA platform available (set 'hibernate.transaction.jta.platform' to enable JTA platform integration)
15:53:02.821 [main] INFO  o.s.o.j.LocalContainerEntityManagerFactoryBean - Initialized JPA EntityManagerFactory for persistence unit 'default'
15:53:03.588 [main] INFO  o.e.j.s.DefaultSessionIdManager - Session workerName=node0
15:53:03.600 [main] INFO  o.e.j.server.handler.ContextHandler - Started osbwej.JettyEmbeddedWebAppContext@6b0aa757{application,/,b=[file:/C:/Users/user/AppData/Local/Temp/jetty-docbase.8080.1297471344677996224/, URLResource@712249DD(jar:nested:///C:/Users/user/Downloads/Stirling-PDF-with-login.exe/!BOOT-INF/lib/swagger-ui-5.2.0.jar!/META-INF/resources/)],a=AVAILABLE,h=oeje10s.SessionHandler@1d0695da{STARTED}}
15:53:03.601 [main] INFO  o.e.j.e.s.ServletContextHandler - Started osbwej.JettyEmbeddedWebAppContext@6b0aa757{application,/,b=[file:/C:/Users/user/AppData/Local/Temp/jetty-docbase.8080.1297471344677996224/, URLResource@712249DD(jar:nested:///C:/Users/user/Downloads/Stirling-PDF-with-login.exe/!BOOT-INF/lib/swagger-ui-5.2.0.jar!/META-INF/resources/)],a=AVAILABLE,h=oeje10s.SessionHandler@1d0695da{STARTED}}
15:53:03.616 [main] INFO  org.eclipse.jetty.server.Server - Started oejs.Server@23ec2bdc{STARTING}[12.0.7,sto=0] @9440ms
15:53:03.651 [main] INFO  s.software.SPDF.SPdfApplication - Running configs ApplicationProperties [security=Security [enableLogin=false, oauth2=null, initialLogin=null, csrfDisabled=true], system=System [defaultLocale=en-US, googlevisibility=false, enableAlphaFunctionality=false, showUpdate=false, showUpdateOnlyAdmin=false], ui=UserInterface [appName=, homeDescription=, appNameNavbar=], endpoints=Endpoints [toRemove=[], groupsToRemove=[CLI]], metrics=Metrics [enabled=true], automaticallyGenerated=AutomaticallyGenerated [key=MASKED], autoPipeline=null]
15:53:04.595 [main] INFO  s.s.S.config.EndpointConfiguration - Disabling file-to-pdf
15:53:04.595 [main] INFO  s.s.S.config.EndpointConfiguration - Disabling repair
15:53:04.597 [main] INFO  s.s.S.config.EndpointConfiguration - Disabling xlsx-to-pdf
15:53:04.597 [main] INFO  s.s.S.config.EndpointConfiguration - Disabling ocr-pdf
15:53:04.597 [main] INFO  s.s.S.config.EndpointConfiguration - Disabling pdf-to-pdfa
15:53:04.598 [main] INFO  s.s.S.config.EndpointConfiguration - Disabling pdf-to-word
15:53:04.599 [main] INFO  s.s.S.config.EndpointConfiguration - Disabling html-to-pdf
15:53:04.599 [main] INFO  s.s.S.config.EndpointConfiguration - Disabling book-to-pdf
15:53:04.599 [main] INFO  s.s.S.config.EndpointConfiguration - Disabling pdf-to-xml
15:53:04.600 [main] INFO  s.s.S.config.EndpointConfiguration - Disabling compress-pdf
15:53:04.600 [main] INFO  s.s.S.config.EndpointConfiguration - Disabling pdf-to-presentation
15:53:04.600 [main] INFO  s.s.S.config.EndpointConfiguration - Disabling url-to-pdf
15:53:04.601 [main] INFO  s.s.S.config.EndpointConfiguration - Disabling pdf-to-book
15:53:04.601 [main] INFO  s.s.S.config.EndpointConfiguration - Disabling pdf-to-rtf
15:53:04.601 [main] INFO  s.s.S.config.EndpointConfiguration - Disabling extract-image-scans
15:53:04.601 [main] INFO  s.s.S.config.EndpointConfiguration - Disabling pdf-to-html
15:53:05.619 [main] WARN  o.s.b.a.o.j.JpaBaseConfiguration$JpaWebConfiguration - spring.jpa.open-in-view is enabled by default. Therefore, database queries may be performed during view rendering. Explicitly configure spring.jpa.open-in-view to disable this warning
15:53:05.816 [main] INFO  o.s.b.a.w.s.WelcomePageHandlerMapping - Adding welcome page: ServletContext resource [/index.html]
15:53:06.242 [main] INFO  o.s.b.a.e.web.EndpointLinksResolver - Exposing 1 endpoint(s) beneath base path '/actuator'
15:53:06.336 [main] INFO  o.s.s.web.DefaultSecurityFilterChain - Will secure any request with [org.springframework.security.web.session.DisableEncodeUrlFilter@40c0151a, org.springframework.security.web.context.request.async.WebAsyncManagerIntegrationFilter@1a57bb34, org.springframework.security.web.context.SecurityContextHolderFilter@6820ab53, org.springframework.security.web.header.HeaderWriterFilter@a9c923f, org.springframework.web.filter.CorsFilter@3e0e2a80, org.springframework.security.web.authentication.logout.LogoutFilter@6178b3bf, stirling.software.SPDF.config.security.UserAuthenticationFilter@7882379b, org.springframework.security.web.savedrequest.RequestCacheAwareFilter@46b7c89b, org.springframework.security.web.servletapi.SecurityContextHolderAwareRequestFilter@1f5faa89, org.springframework.security.web.authentication.AnonymousAuthenticationFilter@4d696bac, org.springframework.security.web.access.ExceptionTranslationFilter@49924fc6, org.springframework.security.web.access.intercept.AuthorizationFilter@3ac09fe2]
15:53:07.101 [main] INFO  o.e.j.s.h.ContextHandler.application - Initializing Spring DispatcherServlet 'dispatcherServlet'
15:53:07.101 [main] INFO  o.s.web.servlet.DispatcherServlet - Initializing Servlet 'dispatcherServlet'
15:53:07.106 [main] INFO  o.s.web.servlet.DispatcherServlet - Completed initialization in 2 ms
15:53:07.142 [main] INFO  o.e.jetty.server.AbstractConnector - Started ServerConnector@5a2661dc{HTTP/1.1, (http/1.1)}{0.0.0.0:8080}
15:53:07.145 [main] INFO  o.s.b.w.e.jetty.JettyWebServer - Jetty started on port 8080 (http/1.1) with context path '/'
15:53:07.168 [main] INFO  s.software.SPDF.SPdfApplication - Started SPdfApplication in 12.014 seconds (process running for 12.993)
15:53:07.178 [scheduling-1] WARN  s.software.SPDF.utils.FileMonitor - not monitoring any directory, even the root directory itself: .\pipeline\watchedFolders
15:53:07.180 [scheduling-1] INFO  s.software.SPDF.utils.FileMonitor - Registered directory: .\pipeline\watchedFolders
15:53:08.198 [main] INFO  s.software.SPDF.SPdfApplication - Stirling-PDF Started.
15:53:08.199 [main] INFO  s.software.SPDF.SPdfApplication - Navigate to http://localhost:8080
test
15:53:28.676 [qtp150385835-52] WARN  s.s.S.c.a.security.RedactController - Invalid color string provided. Using default color BLACK for redaction.
test
Found 0 occurrences of 'test' in the document.
15:53:28.737 [qtp150385835-52] WARN  o.e.j.ee10.servlet.ServletChannel - handleException /api/v1/security/auto-redact java.io.IOException: Maximum size of image exceeded (w * h * scale ^ 2) = 9062.0 * 17500.0 * 4.1666665 ^ 2 > 2147483647

**Reply by buzzman2** [reply]
{
    "Metadata": {
        "Title": null,
        "Author": null,
        "Subject": null,
        "Keywords": null,
        "Producer": null,
        "Creator": null,
        "CreationDate": null,
        "ModificationDate": null
    },
    "FormFields": {},
    "BasicInfo": {
        "FileSizeInBytes": 4267658,
        "WordCount": 0,
        "ParagraphCount": 0,
        "CharacterCount": 6,
        "Compression": false,
        "Language": null,
        "Number of pages": 3
    },
    "DocumentInfo": {
        "PDF version": 1.6,
        "Trapped": null,
        "Page Mode": "USE_NONE"
    },
    "Compliancy": {
        "IsPDF/ACompliant": false,
        "IsPDF/XCompliant": false,
        "IsPDF/ECompliant": false,
        "IsPDF/VTCompliant": false,
        "IsPDF/UACompliant": false,
        "IsPDF/BCompliant": false,
        "IsPDF/SECCompliant": false
    },
    "Encryption": {
        "IsEncrypted": false
    },
    "Other": {
        "EmbeddedFiles": [],
        "Attachments": [],
        "JavaScript": [],
        "Layers": [],
        "Bookmarks/Outline/TOC": [],
        "XMPMetadata": null
    },
    "PerPageInfo": {
        "Page 1": {
            "Size": {
                "Width (px)": "9062.00",
                "Height (px)": "17500.00",
                "Width (in)": "125.86",
                "Height (in)": "243.06",
                "Width (cm)": "319.69",
                "Height (cm)": "617.36",
                "Standard Page": "Custom"
            },
            "Rotation": 0,
            "Page Orientation": "Portrait",
            "MediaBox": "[0.0,0.0,9062.0,17500.0]",
            "CropBox": "[0.0,0.0,9062.0,17500.0]",
            "BleedBox": "[0.0,0.0,9062.0,17500.0]",
            "TrimBox": "Undefined",
            "ArtBox": "[0.0,0.0,9062.0,17500.0]",
            "Text Characters Count": 2,
            "Annotations": {
                "AnnotationsCount": 0,
                "SubtypeCount": 0,
                "ContentsCount": 0
            },
            "Images": [
                {
                    "Width": 9062,
                    "Height": 17500,
                    "ColorSpace": "DeviceRGB"
                }
            ],
            "Links": [],
            "Fonts": [],
            "Color Spaces & ICC Profiles": [],
            "XObjectCounts": {
                "Image": 1
            },
            "Multimedia": []
        },
        "Page 2": {
            "Size": {
                "Width (px)": "9062.00",
                "Height (px)": "17500.00",
                "Width (in)": "125.86",
                "Height (in)": "243.06",
                "Width (cm)": "319.69",
                "Height (cm)": "617.36",
                "Standard Page": "Custom"
            },
            "Rotation": 0,
            "Page Orientation": "Portrait",
            "MediaBox": "[0.0,0.0,9062.0,17500.0]",
            "CropBox": "[0.0,0.0,9062.0,17500.0]",
            "BleedBox": "[0.0,0.0,9062.0,17500.0]",
            "TrimBox": "Undefined",
            "ArtBox": "[0.0,0.0,9062.0,17500.0]",
            "Text Characters Count": 2,
            "Annotations": {
                "AnnotationsCount": 0,
                "SubtypeCount": 0,
                "ContentsCount": 0
            },
            "Images": [
                {
                    "Width": 9062,
                    "Height": 17500,
                    "ColorSpace": "DeviceRGB"
                }
            ],
            "Links": [],
            "Fonts": [],
            "Color Spaces & ICC Profiles": [],
            "XObjectCounts": {
                "Image": 1
            },
            "Multimedia": []
        },
        "Page 3": {
            "Size": {
                "Width (px)": "9062.00",
                "Height (px)": "17500.00",
                "Width (in)": "125.86",
                "Height (in)": "243.06",
                "Width (cm)": "319.69",
                "Height (cm)": "617.36",
                "Standard Page": "Custom"
            },
            "Rotation": 0,
            "Page Orientation": "Portrait",
            "MediaBox": "[0.0,0.0,9062.0,17500.0]",
            "CropBox": "[0.0,0.0,9062.0,17500.0]",
            "BleedBox": "[0.0,0.0,9062.0,17500.0]",
            "TrimBox": "Undefined",
            "ArtBox": "[0.0,0.0,9062.0,17500.0]",
            "Text Characters Count": 2,
            "Annotations": {
                "AnnotationsCount": 0,
                "SubtypeCount": 0,
                "ContentsCount": 0
            },
            "Images": [
                {
                    "Width": 9062,
                    "Height": 17500,
                    "ColorSpace": "DeviceRGB"
                }
            ],
            "Links": [],
            "Fonts": [],
            "Color Spaces & ICC Profiles": [],
            "XObjectCounts": {
                "Image": 1
            },
            "Multimedia": []
        }
    }
}

**Reply by Frooodle** [reply, marked as answer]
yeah sadly our toolkit does not support images with size that large
"Width (px)": "9062.00",
"Height (px)": "17500.00",
(That is 20x larger than a 4k image)

**Comment by buzzman2** [comment]
Thanks. Weird that I did a few months' statements with the same card and
they worked fine.

On Tue, Aug 20, 2024 at 6:13 PM Anthony Stirling ***@***.***>
wrote:

> yeah sadly our toolkit does not support images with size that large
> "Width (px)": "9062.00",
> "Height (px)": "17500.00",
> (That is 20x larger than a 4k image)
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Stirling-Tools/Stirling-PDF/discussions/1605#discussioncomment-10400769>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AXNRBOK3NAWZOLCFEPWIXI3ZSO5OZAVCNFSM6AAAAABLU4OIDOVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTANBQGA3TMOI>
> .
> You are receiving this because you authored the thread.Message ID:
> <Stirling-Tools/Stirling-PDF/repo-discussions/1605/comments/10400769@
> github.com>
>


-- 
*Rick Bergami*
Principal

-- 
"We make technology work for your business."
Tired of worrying about your 
IT?
Click below 
<https://drive.google.com/a/abuzztech.com/file/d/17J2_WaKPMxsIKcel5uwEi63uqCvcWSNW/view?usp=sharing>


 
<https://drive.google.com/a/abuzztech.com/file/d/17J2_WaKPMxsIKcel5uwEi63uqCvcWSNW/view?usp=sharing>






*Full Service Technology Support for your Office. *


*Abuzz 
Technologies*
*(215) 600 - 0349*
*Abuzztech.com*
***@***.*** 
***@***.***>*
